.github/actions/pr-to-update-go/pr_to_update_go/go_pr_maker.py

```diff
@@ -0,0 +1,391 @@
+
+	def set_go_version(self, go_version: str, commit_message: str,
+			source_branch_name: str) -> Commit:
+		"""
+		:param go_version: str
+		:param commit_message: str
+		:param source_branch_name: str
+		:return:
+		:rtype: str
+		"""
+		master: Branch = self.repo.get_branch('master')
+		sha: str = master.commit.sha
+		ref: str = f'refs/heads/{source_branch_name}'
+		self.repo.create_git_ref(ref, sha)
+
+		print(f'Created branch {source_branch_name}')
+		go_version_file: str = self.getenv(ENV_GO_VERSION_FILE)
+		go_file_contents = self.repo.get_contents(go_version_file, ref)
+		kwargs = {'path': go_version_file,
+			'message': commit_message,
+			'content': (go_version + '\n'),
+			'sha': go_file_contents.sha,
+			'branch': source_branch_name,
+		}
+		try:
+			git_author_name = self.getenv(ENV_GIT_AUTHOR_NAME)
+			git_author_email = GIT_AUTHOR_EMAIL_TEMPLATE.format(git_author_name=git_author_name)
+			author: InputGitAuthor = InputGitAuthor(name=git_author_name, email=git_author_email)
+			kwargs['author'] = author
+			kwargs['committer'] = author
+		except KeyError:
+			print('Committing using the default author')
+
+		commit: Commit = self.repo.update_file(**kwargs).get('commit')
+		print(f'Updated {go_version_file} on {self.repo.name}')
+		return commit
+
+	def update_golang_org_x(self, previous_commit: Commit) -> Union[GitCommit, None]:
+		"""
+		:param previous_commit:
+		:type previous_commit:
+		:return:
+		:rtype: Union[GitCommit, None]
+		"""
+		subprocess.run(['git', 'fetch', 'origin'], check=True)
+		subprocess.run(['git', 'checkout', previous_commit.sha], check=True)
+		script_path: str = f'.github/actions/pr-to-update-go/update_golang_org_x.sh'
+		subprocess.run([script_path], check=True)
+		files_to_check: list[str] = ['go.mod', 'go.sum', 'vendor/modules.txt']
+		tree_elements: list[InputGitTreeElement] = []
+		for file in files_to_check:
+			diff_process = subprocess.run(['git', 'diff', '--exit-code', '--', file])
+			if diff_process.returncode == 0:
+				continue
+			with open(file) as stream:
+				content: str = stream.read()
+			tree_element: InputGitTreeElement = InputGitTreeElement(path=file, mode='100644',
+				type='blob', content=content)
+			tree_elements.append(tree_element)
+		if len(tree_elements) == 0:
+			print('No golang.org/x/ dependencies need to be updated.')
+			return
+		tree_hash = subprocess.check_output(
+			['git', 'log', '-1', '--pretty=%T', previous_commit.sha]).decode().strip()
+		base_tree: GitTree = self.repo.get_git_tree(sha=tree_hash)
+		tree: GitTree = self.repo.create_git_tree(tree_elements, base_tree)
+		commit_message: str = f'Update golang.org/x/ dependencies for go{self.latest_go_version}'
+		previous_git_commit: GitCommit = self.repo.get_git_commit(previous_commit.sha)
+		git_commit: GitCommit = self.repo.create_git_commit(message=commit_message, tree=tree,
+			parents=[previous_git_commit],
+			author=self.author, committer=self.author)
+		print(f'Updated golang.org/x/ dependencies')
```

Comment:
This also doesn't need to be an f-string

Reply:
Removed *f* in 978e5f94cd